test/mocha-fixtures.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const mms = require('mongodb-memory-server');
+
+/*
+ * Default MMS mongodb version is used, unless MONGOMS_VERSION is set (which is set with the matrix in test.yml for CI)
+ */
+
+// a single-instance running for single-instance tests
+let mongoinstance;
+
+// a replset-instance for running repl-set tests
+let mongorreplset;
+
+// decide wheter to start a in-memory or not
+const startMemoryInstance = !process.env.MONGOOSE_TEST_URI;
+const startMemoryReplset = !process.env.MONGOOSE_REPLSET_URI;
+
+module.exports.mochaGlobalSetup = async function mochaGlobalSetup() {
+  let instanceuri;
+  let replseturi;
+
+  process.env.RUNTIME_DOWNLOAD = '1'; // ensure MMS is able to download binaries in this context
+
+  // set some options when running in a CI
+  if (process.env.CI) {
+    process.env.MONGOMS_PREFER_GLOBAL_PATH = '1'; // set MMS to use "~/.cache/mongodb-binaries" even when the path does not yet exist
+  }
+
+  if (startMemoryInstance) { // Config to decided if an mongodb-memory-server instance should be used
+    // it's needed in global space, because we don't want to create a new instance every test-suite
+    mongoinstance = await mms.MongoMemoryServer.create({ instance: { args: ['--setParameter', 'ttlMonitorSleepSecs=1'] } });
+    const uri = mongoinstance.getUri();
+    instanceuri = uri.slice(0, uri.lastIndexOf('/'));
+  } else {
+    instanceuri = process.env.MONGOOSE_TEST_URI;
+  }
+
+  if (startMemoryReplset) {
+    mongorreplset = await mms.MongoMemoryReplSet.create({ replSet: { count: 3, args: ['--setParameter', 'ttlMonitorSleepSecs=1'] } }); // using 3 because even numbers can lead to vote problems
+    replseturi = mongorreplset.getUri();
+  } else {
+    replseturi = '';
+  }
+
+  process.env.MONGOOSE_TEST_URI = instanceuri;
```

Comment:
Can we set these as `global` variables rather than overwriting `process.env`? I get why setting these variables is necessary, I just really don't like overwriting `process.env`.

Reply:
i overwrite a `process.env` variable, because that is the exact variable that was already used in `test/common.js`, and so was the easiest way of implementing it (i dont know if `global` carries over, at least in jest it does not)

Reply:
`global` does carry over in Mocha. That is one of many reasons why we advise people against using Jest: https://mongoosejs.com/docs/jest.html#globalsetup-and-globalteardown . Overwriting env var is more of an antipattern than overwriting global IMO, although I think it would be better to just export these.